aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+    // List all governance scripts
+    let scripts = list_scripts(proposal_path)?;
+    if scripts.is_empty() {
+        bail!("no scripts found")
+    }
+
+    println!("Found {} scripts", scripts.len());
+    for path in &scripts {
+        println!("    {}", path.display());
+    }
+
+    // Compile all scripts
+    println!("Compiling scripts...");
+    let mut compiled_scripts = vec![];
+    for path in &scripts {
+        let framework_package_args = FrameworkPackageArgs::try_parse_from([
+            "dummy_executable_name",
+            "--framework-local-dir",
+            &aptos_framework_path().to_string_lossy(),
+            "--skip-fetch-latest-git-deps",
+        ])?;
+
+        let (blob, _hash) = compile_in_temp_dir(
+            "script",
+            path,
+            &framework_package_args,
+            PromptOptions::yes(),
+            None,
+        )?;
+
+        compiled_scripts.push(blob);
+    }
+
+    // Set up the simulation state view
+    let client = Client::new(remote_url);
+    let debugger = AptosDebugger::rest_client(client.clone())?;
+    let state = client.get_ledger_information().await?.into_inner();
+
+    let state_view = SimulationStateView {
+        remote: &debugger.state_view_at_version(state.version),
+        states: Mutex::new(HashMap::new()),
+    };
+
+    // Patch framework functions to skip the governance process
+    print!("Patching framework functions to bypass governance.. ");
+    std::io::stdout().flush()?;
+
+    patch_create_signer(&state_view)?;
+    patch_resolve_multistep_proposal(&state_view)?;
+    println!("done");
+
+    // Create and fund a sender account that is used to send the governance scripts
+    print!("Creating and funding sender account.. ");
+    std::io::stdout().flush()?;
+    let mut rng = aptos_keygen::KeyGen::from_seed([0; 32]);
+    let balance = 100 * 1_0000_0000; // 100 APT
+    let account = AccountData::new_from_seed(&mut rng, balance, 0);
+    state_view.apply_write_set(account.to_writeset());
+    // TODO: should update coin info
+    println!("done");
+
+    // Fetch and update on-chain configs
+    println!("Executing governance scripts...");
+    let gas_schedule = GasScheduleV2::fetch_config(&state_view)
+        .ok_or_else(|| anyhow!("failed to fetch gas schedule v2"))?;
+    let gas_feature_version = gas_schedule.feature_version;
+    let mut gas_params = AptosGasParameters::from_on_chain_gas_schedule(
+        &gas_schedule.into_btree_map(),
+        gas_feature_version,
+    )
+    .map_err(|err| {
+        anyhow!(
+            "failed to construct gas params at gas version {}: {}",
+            gas_feature_version,
+            err
+        )
+    })?;
+    let txn_gas_params = &mut gas_params.vm.txn;
+    // Use the alternative limits for governance proposals
+    // TODO: In the future, consider adding the execution hashes of the scripts to the approval list.
+    txn_gas_params.max_execution_gas = txn_gas_params.max_execution_gas_gov;
```

Comment:
Doesn't your recent change to automatically bump these numbers to `_gov` if it is a governance proposal work here?

Reply:
This is one of the more ugly parts of the current implementation. 

> Doesn't your recent change to automatically bump these numbers to _gov if it is a governance proposal work here?

Yes, but the alt limits will only kick in if the script has its hash added to the list of approved execution hashes, which gets skipped by the mock version of `resolve_multi_step_proposal` as well.

It's possible for us to manually add it in Rust and I tried it, but there are some complexities involved.